pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/ingress_reconciler.go

```diff
@@ -373,10 +373,19 @@ func (ir *IngressReconciler) Reconcile(isvc *v1beta1.InferenceService) error {
 		}
 		if !isvcutils.IsMMSPredictor(&isvc.Spec.Predictor, isvcConfig) {
 			protocol := isvc.Spec.Predictor.GetImplementation().GetProtocol()
+			modelName := isvc.Name
+			// Set model name from env variable for mlserver
+			if isvc.Spec.Predictor.Model != nil {
```

Comment:
Hmm actually each model server has ways to overwrite the model name, not sure how to handle this systematically or maybe we can treat this field as default model url

Reply:
To add additional context to https://github.com/kserve/kserve/issues/2332. I'd like to have different value for isvc.name and the mlserver model name. So I don't think we should add ability to override isvc.name using env. 

The reason to have different value is for having different host and the model_name value in the request payload when calling a model. As of now, it must always be the same. 
E.g if I have isvc.name = "my-model". It will generate host `my-model.<namespace>.<base_url>` and the model payload should contains "model_name" = "my-model" field.  In my case I would like to keep the expected "model_name" in the payload to be the same for several isvc by overriding `MODEL_NAME` env var so that I could introduce some traffic shaping in front of these models.

Reply:
@pradithya here it is overriding the model name in the url path not the host, the host is still using the isvc name, I think that’s what you expected.

Reply:
Yes correct, that's my intention.